components/FinancialSummary/BarChartComponent.tsx

```diff
@@ -30,7 +31,14 @@ export default function BarChartComponent() {
 
     // Initial setup and event listener
     handleResize();
+    const currentYear = new Date().getFullYear();
+
     window.addEventListener('resize', handleResize);
+    import(`../../config/finance/json-data/${currentYear}/ExpensesLink.json`).then((data) =>
+      setExpensesLinkData(data.default)
+    );
+    getUniqueCategories().then((data) => setCategories(data));
+    import(`../../config/finance/json-data/${currentYear}/Expenses.json`).then((data) => setExpensesData(data.default));
```

Comment:
What if Expenses Files of current are not present? In that case this import will lead to error and a deadlock situation will be created. Provide some fallback for these imports.